SQL.Rmd

```diff
@@ -0,0 +1,1190 @@
+| `FOREIGN KEY` | Consistency check that each value in this column corresponds to another value in another table. |
+
+To test this out, let's create a new table called `smra_test`. This will return `NULL` as the query doesn't ask for anything to be returned. Until you add some data, there won't be anything to display either, we'll get to that next. Until then, let's look at our columns:
+
+* `id` - this is just going to be an automatically increasing integer but will be used to uniquely identify the row.
+* `first_name` - while it shouldn't be null, it also shouldn't be longer than 50 characters.
+* `specialty` - this alphanumeric code will not be longer than 4 characters.
+* `los` - (length of stay) is a positive integer.
+
+```{sql create-table, connection="conn", exercise=TRUE}
+
+```
+
+```{sql create-table-solution}
+CREATE TABLE IF NOT EXISTS smra_test (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  first_name CHARACTER(50) NOT NULL,
+  specialty CHARACTER(4),
+  los INTEGER
+);
+```
+
+</br>
+
+### INSERT INTO
+
+When inserting data into a database, we use the `INSERT` statement followed by the name of the table to write to, the columns of data we are filling, and one or more rows of data to insert. If you are relying on default values and not listing every column you need to state the column names explicitly. It's also possible to utilise expressions for the values to insert. 
+
+```sql
+INSERT INTO my_table
+-- optional explicit column listing
+(some_column, another_column, ...)
+VALUES (some_value, another_value...)...;
+```
+
+In our created table, `smra_test`, let's add a couple of rows:
+
+* `first_name`: Harry, `specialty`: A1, `los`: 5
+* `first_name`: Luna, `specialty`: A11, `los`: 7
+* `first_name`: Ron, `specialty`: A2, `los`: 9
+
+```{sql insert-into, connection="conn", exercise=TRUE}
+
+```
+
+```{sql insert-into-solution}
+INSERT INTO smra_test
+(first_name, specialty, LoS)
+VALUES ("Harry", "A1", 5),
+       ("Luna", "A11", 7),
+       ("Ron", "A2", 9);
+```
+
+</br> 
+
+### UPDATE
+
+Depending on your project, it may be a common task to update the data within a table. This is similar to the `INSERT` statement in that you have to be precise with the naming of your table, columns, and also which rows to use the `UPDATE` statement on. Any of the data you are trying to change will still have to match the table constraints from the schema. However, updating data is done with a `WHERE` clause and this is very important or you could end up updating the data across all rows. To avoid this, you could test the query with `SELECT` and ensure you see the right rows before writing the rest of the `UPDATE` statement.
+
+```sql
+UPDATE my_table
+SET column = some_value,
+    another_column = another_value, ...
+WHERE condition(s);
+```
+
+We need to update a couple of things on one of our rows:
```

Comment:
This didn't work for me, similar to above it says smra_test doesn't exist. Also, I don't think the solution matches up to what's being asked?